parsers.js

```diff
@@ -1,137 +1,74 @@
 const xmldoc = require('xmldoc');
 
+const SERVICE_STRING_FIELDS = new Set([
+  'sta',
+  'eta',
+  'std',
+  'etd',
+  'platform',
+  'operator',
+  'operatorCode',
+  'serviceType',
+  'cancelReason',
+  'delayReason',
+  'serviceID',
+  'length',
+  'rsid',
+  'crs',
+  'locationName',
+  'generatedAt',
+]);
+
+const SERVICE_BOOLEAN_FIELDS = new Set([
+  'isCircularRoute',
+  'isCancelled',
+  'filterLocationCancelled',
+  'detachFront',
+  'isReverseFormation',
+  'futureCancellation',
+  'futureDelay',
+]);
+
+const SERVICE_ARRAY_FIELDS = new Set([
+  'origin',
+  'destination',
+  'currentOrigins',
+  'currentDestinations',
+]);
+
 class Parsers {
   static parseArrivalsBoardResponse(soapResponse) {
-    const board = this.getTrainServicesBoard(soapResponse, 'GetArrivalBoardResponse');
-    const trains = [];
-
-    try {
-      board.eachChild((service) => {
-        trains.push(this.parseStandardService(service));
-      });
-    } catch (e) {
-      console.error(e);
-    }
-
-    return { trainServices: trains };
+    return this.getTrainServicesBoard(soapResponse, 'GetArrivalBoardResponse');
```

Comment:
These are all very similar, and I was able to combine them into a single service board method.